underdocs-renderer/src/main/kotlin/underdocs/renderer/output/html/render/page/PageRenderer.kt

```diff
@@ -57,8 +60,26 @@ class PageRenderer(private val linker: Linker,
                             .withHref(linker.siteLinkBetween(visitable, "_static/code.css"))
             ),
             body(
+                    script()
+                            .with(UnescapedText("""
+                                (function () {
+                                  if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
```

Comment:
Since this condition is used twice on the same page, I think it would be a good idea to extract it into a predicate function or a global variable. Something like this:

~~~~JavaScript
const isNightModePreferred =  () => window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
~~~~

Also, [optional chaining (?.)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) seems to have good support (according to [caniuse](https://caniuse.com/#search=optional%20chaining), it is >75%), thus you can even simplify this expression:


~~~~JavaScript
const isNightModePreferred =  () => window.matchMedia?.('(prefers-color-scheme: dark)').matches;
~~~~

Reply:
Good idea, optional chaining looks nice! Created global variable for the condition.